controllers/mirrorpeersecret_controller.go

```diff
@@ -97,5 +102,34 @@ func mirrorPeerSecretReconcile(ctx context.Context, rc client.Client, req ctrl.R
 func (r *MirrorPeerSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Secret{}, builder.WithPredicates(utils.SourceOrDestinationPredicate)).
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(r.secretConfigMapFunc)).
 		Complete(r)
 }
+
+func (r *MirrorPeerSecretReconciler) secretConfigMapFunc(obj client.Object) []reconcile.Request {
+	ConfigMapRamenConfigKeyName := "ramen_manager_config.yaml"
+	var cm *corev1.ConfigMap
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return []reconcile.Request{}
+	}
+
+	ramenConfig := &ramendrv1alpha1.RamenConfig{}
+	err := yaml.Unmarshal([]byte(cm.Data[ConfigMapRamenConfigKeyName]), ramenConfig)
+	if err != nil {
+		return []reconcile.Request{}
+	}
```

Comment:
Does this work if the key doesn't exist in ConfigMap? or, will it panic?

Reply:
I am not sure about this so i've added a check to see if the key exists first